Python/Product/Uwp/Project/PythonUwpProject.cs

```diff
@@ -64,9 +67,33 @@ partial class PythonUwpProject :
         /// information from the project)
         /// </summary>
         protected override void InitializeForOuter(string fileName, string location, string name, uint flags, ref Guid guidProject, out bool cancel) {
+            var pythonProject = this.GetProject().GetPythonProject();
+            var msbuildProject = pythonProject.GetMSBuildProjectInstance();
+            msbuildProject.Build("CreatePythonUwpIoTPythonEnv", null);
+
+            var sitePackagesDir = Path.Combine(pythonProject.ProjectDirectory, PythonUwpConstants.InterpreterRelativePath, PythonUwpConstants.InterpreterLibPath, "site-packages");
+            var sitePackageDirInfo = new DirectoryInfo(sitePackagesDir);
+            if (sitePackageDirInfo.Exists) {
+                sitePackageWatcher = new FileSystemWatcher {
+                    IncludeSubdirectories = true,
+                    Path = sitePackagesDir,
+                };
+
+                sitePackageWatcher.Created += SitePackageWatcher_Changed;
+                sitePackageWatcher.Changed += SitePackageWatcher_Changed;
+                sitePackageWatcher.Deleted += SitePackageWatcher_Changed;
+                sitePackageWatcher.Renamed += SitePackageWatcher_Changed;
+                sitePackageWatcher.EnableRaisingEvents = true;
+            }
+
             base.InitializeForOuter(fileName, location, name, flags, ref guidProject, out cancel);
         }
 
+        private void SitePackageWatcher_Changed(object sender, System.IO.FileSystemEventArgs e) {
+            var bps = this._innerProject as IVsBuildPropertyStorage;
```

Comment:
"IIRC this event is raised off the UI thread, so you may just want to send an action to the UI thread to update the property."  -- can you explain this one?  Changing msbuild properties need to be on the UI thread?